### DISCUSSION TITLE
Please add a method to the C# library to copy a directory with all its contents

### DISCUSSION THREAD

**Discussion by aks2161989** [dotnet/csharplang#5456] [Language Ideas] [answered]
Currently C# does not have an inbuilt method to copy a directory and all its contents. To achieve this task, one needs to write code for it. [An example is given on the MSDN website](https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo?redirectedfrom=MSDN&view=net-6.0).

What is strange, however, is that [VB.NET already has an inbuilt method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualbasic.fileio.filesystem.copydirectory?view=net-6.0) which allows you to achieve this task in one line. Sure, one can add a reference to `Microsoft.VisualBasic` in their C# project and call the VB.NET method, but I am sure most C# developers would appreciate if such a method were to be incorporated in the language itself.

What's more is that C# has long tried to compete with Java and has succeeded in many aspects. However, [Java already has inbuilt methods to copy a directory and its contents](https://stackoverflow.com/a/1146651/6623360) and does not require extra work to achieve this. If such a feature were to be added to C#, it would definitely enhance the competitiveness of the language.

**Comment by HaloFour** [comment, marked as answer]
Requests for new APIs in the .NET class library can be made in the [runtime repository](https://github.com/dotnet/runtime).

**Reply by aks2161989** [reply]
Thank you. I will repost it there
